### PR TITLE
I have added CTI support in this fork

### PR DESCRIPTION
**Pull request by vasil-yordanov**

### CODE REVIEW

File: src/SimpleThings/EntityAudit/AuditReader.php [hunk 1]
```diff
@@ -153,215 +53,63 @@ public function __construct(EntityManager $em, AuditConfiguration $config, Metad
...
-        $options = array_merge(
-            array(
-                'threatDeletionsAsExceptions' => false
-            )
-            , $options
-        );
-
         if (!$this->metadataFactory->isAudited($className)) {
-            throw new NotAuditedException($className);
+            throw AuditException::notAudited($className);
```
Comment (smoench) [inline src/SimpleThings/EntityAudit/AuditReader.php]: @vasil-yordanov You may have forgotten to add  the `AuditException` file.

Comment (vasil-yordanov) [inline src/SimpleThings/EntityAudit/AuditReader.php]: Hi,

Isn't this the same as

new NotAuditedException($className);

?

Have you merged my branch to master?

Regards,

Vasil.

Comment (DavidBadura) [inline src/SimpleThings/EntityAudit/AuditReader.php]: No, `NotAuditedException` isn't the same name as `AuditException` :wink:  Do you forget a use statement or add the class?

The builds are failed, see https://travis-ci.org/simplethings/EntityAudit/builds/69906870

```
PHP Fatal error:  Class 'SimpleThings\EntityAudit\AuditException' not found in /home/travis/build/simplethings/EntityAudit/src/SimpleThings/EntityAudit/AuditReader.php on line 411
```

Comment (vasil-yordanov) [inline src/SimpleThings/EntityAudit/AuditReader.php]: ok I will fix this tomorrow.

Regards,

Vasil

Comment (vasil-yordanov) [inline src/SimpleThings/EntityAudit/AuditReader.php]: Hello, I'm very sorry for my late response.
I fixed this issue.
Please let me know if there is still some chance to include my changes in
your code.

Regards,

Vasil.

**Comment by DavidBadura** [comment]
it has merge conflicts - can you it rebase please?

**Comment by vasil-yordanov** [comment]
Ok, I will do it now

**Comment by vasil-yordanov** [comment]
Hello David,

It seems this is not a trivial merge I will need more time to do that, will
be not able to do the merge and the testing today. I hope will manage in
2-3 days to do that.

Regards,

Vasil.

**Comment by OskarStark** [comment]
great work :+1:

**Comment by DavidBadura** [comment]
no problem. i would like to merge it before #159. thank you!

**Comment by vasil-yordanov** [comment]
Hello David,

I give up to do the merge, its too complex.
It is a pity because I did a lot of code refactoring in my version where I
managed to shorten the methods. Currently the code has very long methods.
See my old code:
https://github.com/vasil-yordanov/EntityAudit/blob/4b5a6a4b9f37c6276f85d57b96d2d86a9f0a93a5/src/SimpleThings/EntityAudit/AuditReader.php

It seems that currently the feature (excluding the code refactoring) that I
have added previously in my repo is already available in the master of
simplethings/EntityAudit

However in my latest commit I did the following:
1. I have merged my code with the master of simplethings/EntityAudit repo.
2. I added some functional tests for Joined table inheritance.
3. I fixed some small bugs found during the tests
4. I did very small code refactoring.

I will try some shiny day to continue with the refactoring of the
code, but *please
accept now my pull request,* and update the readme that we this lib
supports Joined table inheritance.

Regards,

Vasil.

**Comment by smoench** [comment]
@vasil-yordanov Could you please open a new clean PR with your latest changes? Thanks in advance

**Pull request closed without merging**